addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,281 @@
+        Searching the e-invoice information from Ecpay API and update the invoice information in Odoo
+        """
+        self.ensure_one()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+
+        json_data = {
+            "MerchantID": self.company_id.l10n_tw_edi_ecpay_merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+        }
+
+        response_data = EcPayAPI(self.company_id).call_ecpay_api("/GetIssue", json_data)
+        if response_data.get('RtnCode') != 1:
+            return [_("Invoice '%(name)s' Error: '%(error_message)s'", name=self.name, error_message=response_data.get("RtnMsg"))]
+
+        self.l10n_tw_edi_state = "valid" if response_data.get("IIS_Invalid_Status") == "0" else "invalid"
+
+    def _l10n_tw_edi_run_invoice_invalid(self):
+        """
+        Cancelling the e-invoice by calling the Ecpay API and update the invoice information in Odoo
+        """
+        self.ensure_one()
+        if not self.l10n_tw_edi_ecpay_invoice_id:
+            raise UserError(_("You cannot invalidate an invoice that was not sent to Ecpay."))
+        if self.l10n_tw_edi_state == "invalid":
+            raise UserError(_("The invoice: '%(invoice_id)s' has already been invalidated", invoice_id=self.l10n_tw_edi_ecpay_invoice_id))
+
+        json_data = {
+            "MerchantID": self.company_id.l10n_tw_edi_ecpay_merchant_id,
+            "InvoiceNo": self.l10n_tw_edi_ecpay_invoice_id,
+            "InvoiceDate": self.l10n_tw_edi_invoice_create_date.strftime("%Y-%m-%d %H:%M:%S"),
+            "Reason": self.name
+        }
+
+        response_data = EcPayAPI(self.company_id).call_ecpay_api("/Invalid", json_data)
+
+        if response_data.get("RtnCode") != 1:
+            raise UserError(_("Fail to invalidate invoice. Error message: '%(error_message)s'", error_message=response_data.get("RtnMsg")))
+
+        # update the invoice information in Odoo
+        self._l10n_tw_edi_update_ecpay_invoice_info()
+
+    def l10n_tw_edi_issue_allowance(self):
+        """
+        Issuing an allowance by calling the Ecpay API and update the refund invoice information in Odoo
+        Two methods to issue the allowance
+        1. Endpoint: /Allowance
+            General allowance, which requires merchants or sellers to get the agreement from the customer first (not by using ECPay's system)
+            and then to send an API request to ECPay to issue an allowance.
+        2. Endpoint: /AllowanceByCollegiate
+            Sending an API request to ECPay and ECPay will send an e-mail notification with a link to the customer to get his/her agreement
+            ONce the customer clicks the link, an allowance will be issued instantly
+        """
+        self.ensure_one()
+        if not self.l10n_tw_edi_ecpay_invoice_id:
+            raise UserError(_("You cannot refund an invoice that was not sent to Ecpay."))
```

Comment:
```suggestion
            raise UserError(_("You cannot issue an allowance for an invoice that was not sent to Ecpay."))
```
Also maybe provide the invoice number in the message to be as clear as you can